test/Dialect/FIRRTL/lint.mlir

```diff
@@ -0,0 +1,45 @@
+// RUN: circt-opt --pass-pipeline='builtin.module(firrtl.circuit(firrtl.module(firrtl-lint)))' %s | FileCheck %s
+
+firrtl.circuit "lint_tests" {
+  // CHECK: firrtl.module @lint_tests
+  firrtl.module @lint_tests(in %en: !firrtl.uint<1>, in %pred: !firrtl.uint<1>, in %reset: !firrtl.reset, in %clock: !firrtl.clock) {
+    %0 = firrtl.asUInt %reset : (!firrtl.reset) -> !firrtl.uint<1>
+    // CHECK: firrtl.assert
+    firrtl.assert %clock, %pred, %en, "valid" : !firrtl.clock, !firrtl.uint<1>, !firrtl.uint<1>  {eventControl = 0 : i32, isConcurrent = false}
+    // CHECK: firrtl.assert
+    firrtl.assert %clock, %0, %en, "valid" : !firrtl.clock, !firrtl.uint<1>, !firrtl.uint<1>  {eventControl = 0 : i32, isConcurrent = false}
+    %false = firrtl.constant 0 : !firrtl.uint<1>
+    // CHECK: firrtl.assert
+    firrtl.assert %clock, %false, %en, "valid" : !firrtl.clock, !firrtl.uint<1>, !firrtl.uint<1>  {eventControl = 0 : i32, isConcurrent = false}
+    // CHECK: firrtl.int.verif.assert
+    firrtl.int.verif.assert %pred : !firrtl.uint<1>
+  }
+
+  // expected-error @below {{'firrtl.assert' op is guarunteed to fail simulation, as the predicate is constant false}}
+  firrtl.module @assert_const(in %clock: !firrtl.clock) {
+    %true = firrtl.constant 1 : !firrtl.uint<1>
+    %false = firrtl.constant 0 : !firrtl.uint<1>
+    firrtl.assert %clock, %false, %true, "valid" : !firrtl.clock, !firrtl.uint<1>, !firrtl.uint<1>  {eventControl = 0 : i32, isConcurrent = false}
+  }
+
+  // expected-error @below {{'firrtl.assert' op is guarunteed to fail simulation, as the predicate is a reset signal}}
```

Comment:
Maybe add an expected-note for the reset note expectation?

Reply:
Thanks, missed them.